### DISCUSSION TITLE
关于添加元数据readYear 和 tag 标签

### DISCUSSION THREAD

**Discussion by mockingtao** [zhaohongxuan/obsidian-weread-plugin#264] [Q&A]
问题1：元数据中我看现在没有现成的readYear 字段，准备通过 metadata.lastReadDate 做一步转化，但是似乎语法不大对。求助正确写法。代码如下：
`readYear: {{ metaData.lastReadDate | date('YYYY') }}`

问题2：想对元数据中添加tag字段，且tag想为metadata.category的"-"字符前面的部分，还是语法问题，不大对，求助正确写法。代码如下:
`tag: {{ metaData.category | split('-')[0] }}`

**Comment by ZiGmaX809** [comment]
nunjunks语法太拉了，想要获取year和tag标签的目的是在dataview中展示么？  你还是用dataview的js功能吧

**Reply by irisfish531** [reply]
JS功能不会了

**Reply by ZiGmaX809** [reply]
参考我的仓库

[ReadingCard.js](https://raw.githubusercontent.com/ZiGmaX809/obsidian-readingcard-template/main/ReadingCard.js)

**Reply by mockingtao** [reply]
感谢，dataviewjs 试了下可以的。

**Comment by irisfish531** [comment]
有同样的问题，也是想用READYEAR获取年份。

**Reply by mockingtao** [reply]
我后来是直接用dataview ，可以实现一个还不错的列表：
见源码：
```
dataview
table without id
	("![]("+ cover + ")") as Cover, 
	file.link as Name , 
	author as Author, 
	readingStatus,
	progress,
	dateformat(lastReadDate, "yyyy-MM-dd") as LastRead
from "这里是微信读书同步的文件夹路径"  
where cover != null and date(lastReadDate).year = date(today).year
sort readingStatus desc, lastReadDate desc
```

**Reply by irisfish531** [reply]
我想显示本月读的书的话，WHRER语句改为了“	where cover != null and date(lastReadDate).year = date(today).year and date(lastReadDate).month = date(today).month” ，但啥都不显示。。。求指点

**Reply by irisfish531** [reply]
`where cover != null and date(lastReadDate).year = date(today).year and date(lastReadDate).month = date(today).month`

**Reply by mockingtao** [reply]
我这边按照上面语句是可以的。有没有可能你还没有拉取最新的微信读书记录？
